File: ios-native-shell-scaffold/VineSightShell/BootstrapRouter.swift
import Foundation

enum BootstrapAuthState {
    case loading
    case unauthenticated
    case profileIncomplete
    case authenticated
}

struct BootstrapSnapshot {
    let isLoading: Bool
    let isAuthenticated: Bool
    let needsProfileCompletion: Bool
    let hasProfileName: Bool
    let isProfileLoading: Bool
}

enum BootstrapRouter {
    static func resolve(_ snapshot: BootstrapSnapshot) -> NativeRouteId? {
        if snapshot.isLoading || (snapshot.isAuthenticated && snapshot.isProfileLoading) {
            return nil
        }
        if !snapshot.isAuthenticated {
            return .authPhoneLogin
        }
        if snapshot.needsProfileCompletion || !snapshot.hasProfileName {
            return .authProfileCompletion
        }
        return .tabsHome
    }
}

File: android-native-shell-scaffold/app/src/main/java/com/vinesight/shell/BootstrapRouter.kt
package com.vinesight.shell

data class BootstrapSnapshot(
    val isLoading: Boolean,
    val isAuthenticated: Boolean,
    val needsProfileCompletion: Boolean,
    val hasProfileName: Boolean,
    val isProfileLoading: Boolean,
)

object BootstrapRouter {
    fun resolve(snapshot: BootstrapSnapshot): NativeRouteId? {
        if (snapshot.isLoading || (snapshot.isAuthenticated && snapshot.isProfileLoading)) {
            return null
        }
        if (!snapshot.isAuthenticated) {
            return NativeRouteId.AUTH_PHONE_LOGIN
        }
        if (snapshot.needsProfileCompletion || !snapshot.hasProfileName) {
            return NativeRouteId.AUTH_PROFILE_COMPLETION
        }
        return NativeRouteId.TABS_HOME
    }
}
